Adding an implementation of simpleion.dumps

Antrikshy opened the pull request:
I implemented `simpleion.dumps` as a simple wrapper over `simpleion.dump` using `six.BytesIO` as the file buffer.

I got an error when running existing tests:

    tests/test_vectors.py:328: in <module>
        _basic_params(_T.BAD, _BAD_UTF8_SUBDIR),
    tests/test_vectors.py:199: in _basic_params
        for file in _list_files(directory_path):
    tests/test_vectors.py:174: in _list_files
        for file in listdir(directory_path):
    E   OSError: [Errno 2] No such file or directory: '/path/to/ion-python/vectors/iontestdata/good'

Does it have to do with my change or something I'm missing in my dev environment?

I hope contributions are accepted!

tgregg commented:
@Antrikshy Thanks for the contribution. Did you initialize the ion-tests submodule?

`git submodule init`
`git submodule update`

Antrikshy commented:
Thanks! I just did, and the tests run fine. I added a simple test runner for my new method as well.

I also noticed that the existing stub for `dumps` was missing a couple of arguments. I ignored them earlier because I didn't know what they did and thought it was intentional. It caused some test cases to fail, so I added them in.

tgregg commented on amazon/ion/simpleion.py:
```diff
@@ -203,12 +204,54 @@ def _dump(obj, writer, field=None):
+
+    Args:
+        obj (Any): A python object to serialize according to the above table. Any Python object which is neither an
+            instance of or inherits from one of the types in the above table will raise TypeError.
```
or -> nor

tgregg commented on tests/test_simpleion.py:
```diff
@@ -318,6 +318,23 @@ def equals():
+    # test dumps
+    res = dumps(p.obj, sequence_as_stream=p.stream)
+    if not p.has_symbols:
+        assert (b'$ion_1_0 ' + p.expected) == res
```
Per the [Travic CI build](https://travis-ci.org/amzn/ion-python/builds/285880665?utm_source=github_status&utm_medium=notification), this fails on Python 3, probably because `res` is a text (unicode) type while the left hand side is a binary type (bytes). You may need to decode the left hand side.

The [README](https://github.com/amzn/ion-python/blob/master/README.md) contains instructions for setting up `tox` to run local builds for multiple python interpreter versions.

Antrikshy replied:
You're right. Decoding the left side resolves it.

As the README says, I'm having some issues getting pyenv to do things on macOS, but I was able to run tests manually on 2.7.10 and 3.6.3 installations and they're all green. I'll let Travis CI do its thing.
